### ISSUE TITLE
`get_array_statistics` fails when data is fully masked 

### ISSUE THREAD

**Issue by vincentsarago** [cogeotiff/rio-tiler#617]
First reported in https://github.com/NASA-IMPACT/veda-data-pipelines/issues/350

```python
import numpy
from rio_tiler.utils import get_array_statistics

arr = numpy.ma.array(numpy.zeros((1,256,256)))
arr.mask = True
get_array_statistics(arr)

IndexError: cannot do a non-empty take from an empty axes
```

We will fix this in rio-tiler v4 and V5

**Comment by vincentsarago** [comment]
🤦 this was already fixed in 5.0 https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/utils.py#L117-L122

**Comment by vincentsarago** [comment]
closed in #618
